### ISSUE TITLE
Support unittest.setUpClass

### ISSUE THREAD

**Issue by PeterJCLaw** [python/mypy#8723]
Mypy supports adding items to `self` during test `setUp`, which is great. However in many cases it's useful to be able to add items to the class during `setUpClass` (or Django's `setUpTestData`).

```python
import unittest

class Tests(unittest.TestCase):
    @classmethod
    def setUpClass(cls) -> None:
        cls.foo = 42  # "Type[Tests]" has no attribute "foo"

    def test_foo(self) -> None:
        print(self.foo)  # "Tests" has no attribute "foo"
```

Adding a `ClassVar` allows for a workaround to this, however it would be nice if that wasn't needed.

**Comment by msullivan** [comment]
Yeah, adding to a class through a class variable write could be handy, though I doubt it is really on the roadmap for us to implement. There ought to be an issue open for it but I couldn't find one, so I'll leave this open.

**Comment by PeterJCLaw** [comment]
Ah, I'd assumed that the support for `setUp` was specific to unittest handling. Is it just general handling around adding properties to instances? (I had thought that `mypy` would normally error if you tried to add properties to instances)

**Comment by rbharadwaj9** [comment]
@PeterJCLaw can you elaborate on your workaround to this problem? I tried typing my variables in `setUpClass` with `ClassVar` but it doesn't seem to help. I think I'm going it wrong.

**Comment by PeterJCLaw** [comment]
I think I worked around this by adding the variables to the class and marking them as `ClassVar` there.
For example see the variable `srcomp_instance` in the following: https://github.com/PeterJCLaw/srcomp/blob/d3839d4578728f17a052ce934e8f6fe05a38dfad/tests/test_comp.py#L13-L19

**Comment by rbharadwaj9** [comment]
That worked for me. Thanks!
